docs/usage/code-coverage.mdx

```diff
@@ -7,121 +7,136 @@ description: Pester can generate code coverage metrics to help you understand ho
-        FunctionOne | Should -Be 'SwitchParam was not set'
-    }
+   ```powershell
+   # Create a Pester configuration object using `New-PesterConfiguration`
+   $config = New-PesterConfiguration
+   
+   # Set the test path to specify where your tests are located. In this example, we set the path to the current directory. Pester will look into all subdirectories.
+   $config.Run.Path = "."
 
-    It 'Calls FunctionTwo' {
-        FunctionTwo | Should -Be 'I get executed'
-    }
-}
-```
+   # Enable Code Coverage
+   $config.CodeCoverage.Enabled = $true
 
-As you can see, the test script fails to run `FunctionOne` with its switch parameter set, and there is an unreachable line of code in `FunctionTwo`. Here are the results of calling `Invoke-Pester` with different values passed to its `-CodeCoverage` parameter:
+   # Run Pester tests using the configuration you've created
+   Invoke-Pester -Configuration $config
 
-```powershell
-Invoke-Pester .\CoverageTest.Tests.ps1 -CodeCoverage .\CoverageTest.ps1
+   # Example output...
+   # Tests completed in 8.26s
+   # Tests Passed: 8, Failed: 0, Skipped: 0 NotRun: 0
+   # Processing code coverage result.
+   # Covered 11.7% / 75%. 735 analyzed Commands in 22 Files.
+   ```
+
+The final line displays both the current code coverage and the desired target coverage. You can customize the target coverage by configuring the CodeCoverage.CoveragePercentTarget option.
+
+## Coverage Format
 
-<#
-Code coverage report:
-Covered 60.00 % of 5 analyzed commands in 1 file.
+By default, Pester generates a `coverage.xml` file in [JaCoCo](https://www.jacoco.org/) format. You have the flexibility to modify the output format, path, and encoding by using the following options:
 
-Missed commands:
+```powershell
+$config = New-PesterConfiguration
+$config.CodeCoverage.OutputFormat = 'CoverageGutters'
+$config.CodeCoverage.OutputPath = 'cov.xml'
+$config.CodeCoverage.OutputEncoding = 'UTF8'
 
-File             Function    Line Command
-----             --------    ---- -------
-CoverageTest.ps1 FunctionOne    5 return 'SwitchParam was set'
-CoverageTest.ps1 FunctionTwo   16 return 'I do not'
-#>
+$config.CodeCoverage.Enabled = $true
 
-Invoke-Pester .\CoverageTest.Tests.ps1 -CodeCoverage @{Path = '.\CoverageTest.ps1'; Function = 'FunctionOne' }
+Invoke-Pester -Configuration $config
+```
 
-<#
-Code coverage report:
-Covered 66.67 % of 3 analyzed commands in 1 file.
+For additional `CodeCoverage` configuration options, please refer to the [New-PesterConfiguration](https://pester.dev/docs/commands/New-PesterConfiguration) documentation.
 
-Missed commands:
+> **Note:** Pester does not traverse directories. 
+If you encounter an **empty `coverage.xml` file**, it's likely because you're running the tests from a directory that doesn't
+include the relevant code to be covered. 
+To resolve this, you can either follow the recommended [Test file structure](https://pester.dev/docs/usage/test-file-structure),
```

Comment:
There's a `CodeCoverage.Path` that should be prefered. A parent folder may include other files like build/release/test scripts etc. that you don't want to include.

Reply:
This is cool, thanks for the suggestion.